From 9648c5ea5a9c740db6df8d590945c7a68ee371d4 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 29 Mar 2022 13:58:22 +0300
Subject: [PATCH] [SW-77251] common: rename recovery progress bit

Since ROM performs multiple operations in recovery,
this patch renames progress bit accordingly.

Change-Id: I6f352b2409860dc0faafa26718b2a7f6b18e2f57
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 970fcad..c2793ff 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -24,8 +24,6 @@
 
 static void rom_pcie_recovery(void)
 {
-	arcboot_rom_status_update(PROGRESS_PCIE_IN_RECOVERY);
-
 	/* Disable PCIe error responses (both HBW and LBW) */
 	regmap_writel(0x1, mmPCIE_RSP_OK);
 	regmap_writel(0x1, mmD0_PCIE_WRAP_PCIE_RSP_OK_1);
@@ -45,6 +43,8 @@
 {
 	uint8_t recovery_location;
 
+	arcboot_rom_status_update(PROGRESS_ARCBOOT_IN_RECOVERY);
+
 	/* Notify slave if master die and dual die sku */
 	if ((arcboot_rom_global.sku) && (!arcboot_rom_global.die_id))
 		arcboot_rom_notify_slave();
diff --git a/common/pcie.c b/common/pcie.c
index d2cb969..2c36b25 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -145,7 +145,7 @@
 	if ((arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SPI_CONTROLLER) != 0)
 		spi_en = 1;
 
-	arcboot_rom_status_update(PROGRESS_PCIE_IN_RECOVERY);
+	arcboot_rom_status_update(PROGRESS_ARCBOOT_IN_RECOVERY);
 
 	/*
 	 * check and enable 4B termination
diff --git a/include/arcboot.h b/include/arcboot.h
index e819c30..5d6d2a0 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -157,7 +157,7 @@
 	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 48,
 	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 49,
 	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 50,
-	PROGRESS_PCIE_IN_RECOVERY = 51,
+	PROGRESS_ARCBOOT_IN_RECOVERY = 51,
 	PROGRESS_PPBOOT_START = 52,
 	PROGRESS_PPBOOT_FINISH = 53,
 	PROGRESS_BRINGUP_RAM_PRI = 54,
